task: flaky test: TestScheduler_UpdateTask

TestScheduler_UpdateTask is flaky. I've bisected to fca6a9c41297e1b5fe129cd148d09dce39dbebc2. On that commit, `go test -run=Scheduler_UpdateTask -failfast -count=10  ./task/backend/` appears to always fail. Running with count of 1 passes sometimes.

The output during failure looks like:

```
--- FAIL: TestScheduler_UpdateTask (0.22s)
    <autogenerated>:1: 2019-03-16T04:28:52.678Z	INFO	Created run; beginning execution	{"svc": "taskd/scheduler", "task_id": "0000000000000001", "run_slot": 0, "run_id": "0000000000000001", "now": 3060}
    <autogenerated>:1: 2019-03-16T04:28:52.687Z	DEBUG	Ticked	{"svc": "taskd/scheduler", "now": 3060, "tasks_affected": 1}
    <autogenerated>:1: 2019-03-16T04:28:52.687Z	DEBUG	Ticked	{"svc": "taskd/scheduler", "now": 3061, "tasks_affected": 0}
    <autogenerated>:1: 2019-03-16T04:28:52.687Z	INFO	Failed to finish run	{"svc": "taskd/scheduler", "task_id": "0000000000000001", "run_slot": 0, "run_id": "0000000000000001", "now": 3060, "error": "unknown run ID 0000000000000001; known run IDs: "}
    <autogenerated>:1: 2019-03-16T04:28:52.687Z	DEBUG	Ticked	{"svc": "taskd/scheduler", "now": 3600, "tasks_affected": 1}
    scheduler_test.go:310: did not see count of 1 running task(s) for ID 0000000000000001 in time; last count was 0
```

On the parent commit 0d045630c, `go test -run=Scheduler_UpdateTask -failfast -count=100  ./task/backend/` doesn't appear to ever fail.

I'm skipping the test for now.

--- a/task/backend/scheduler.go
+++ b/task/backend/scheduler.go
@@ -348,6 +348,21 @@ func (s *TickScheduler) UpdateTask(task *StoreTask, meta *StoreTaskMeta) error {
 	ts.nextDue = next
 	ts.nextDueMu.Unlock()
 
+	// check the concurrency
+	// todo(lh): In the near future we may not be using the scheduler to manage concurrency.
+	maxC := int(meta.MaxConcurrency)
+	if maxC != len(ts.runners) {
+		if maxC < len(ts.runners) {
+			ts.runners = ts.runners[:maxC]
+		}
+
+		if maxC > len(ts.runners) {
+			delta := maxC - len(ts.runners)
+			for i := 0; i < delta; i++ {
+				ts.runners = append(ts.runners, newRunner(s.ctx, ts.wg, s.logger, task, s.desiredState, s.executor, s.logWriter, ts))
+			}
+		}
+	}
 	if now := atomic.LoadInt64(&s.now); now >= next || hasQueue {
 		ts.Work()
 	}

--- a/task/backend/scheduler_test.go
+++ b/task/backend/scheduler_test.go
@@ -260,7 +260,6 @@ func TestScheduler_Release(t *testing.T) {
 }
 
 func TestScheduler_UpdateTask(t *testing.T) {
-	t.Skip("flaky test: https://github.com/influxdata/influxdb/issues/12667")
 	t.Parallel()
 
 	d := mock.NewDesiredState()

